### PR TITLE
Add window decoration buttons for Wayland

### PR DESCRIPTION
**Pull request by HidenoriMatsubayashi**
I added simple close/max/min buttons for Wayland backend. However, there are some issues with this features (e.g. min/max buttons don't work correctly on some Wayland compositors like Weston)

decorations

## Related issues
- https://github.com/sony/flutter-embedded-linux/issues/165
- https://github.com/sony/flutter-embedded-linux/issues/6

## Related PRs
- https://github.com/sony/flutter-embedded-linux/pull/180

### CODE REVIEW

**Comment by HidenoriMatsubayashi** [comment]
@taka-okayama Can you review my PR?

File: src/flutter/shell/platform/linux_embedded/surface/environment_egl.h [hunk 1]
```diff
@@ -28,8 +28,8 @@ class EnvironmentEgl {
 
   EnvironmentEgl() : display_(EGL_NO_DISPLAY), valid_(false) {}
```
Comment (taka-okayama) [inline src/flutter/shell/platform/linux_embedded/surface/environment_egl.h:29]: sub_environment_(false) is missing. Is this intentional?

File: src/flutter/shell/platform/linux_embedded/surface/environment_egl.h [hunk 2]
```diff
@@ -28,8 +28,8 @@ class EnvironmentEgl {
...
 
-  ~EnvironmentEgl() {
-    if (display_ != EGL_NO_DISPLAY) {
+  virtual ~EnvironmentEgl() {
```
Comment (taka-okayama) [inline src/flutter/shell/platform/linux_embedded/surface/environment_egl.h:31]: If all destructors in the derived class are 'default' and there are no other virtual functions, I don't think you need to make this a virtual destructor.

**Review by taka-okayama — APPROVED** [review]
I think it's OK.

However, I did make a few minor comments, but I'll leave it up to you to decide if you want to fix them or not.

File: src/flutter/shell/platform/linux_embedded/surface/environment_egl.h (hunk 1 above)
Comment (HidenoriMatsubayashi) [reply src/flutter/shell/platform/linux_embedded/surface/environment_egl.h]: Ultimately I deleted EnvironmentEglSub class and refactored source files.

File: src/flutter/shell/platform/linux_embedded/surface/environment_egl.h (hunk 2 above)
Comment (HidenoriMatsubayashi) [reply src/flutter/shell/platform/linux_embedded/surface/environment_egl.h]: Deleted virtual.

**Pull request merged**